### PR TITLE
Add failing unit test for empty string locale

### LINKED ISSUE #135: Translate doesn't revert to default locale when passing empty string as locale
### BC Break Report

|    Q        |   A
|------------ | ------
| Version     | 2.27.0

#### Summary

Passing empty string used to be the same as passing null as locale to `\Laminas\I18n\Translator\Translator::translate()` but the behavior changed.

```diff
@@ -345,7 +345,7 @@ public function getPluginManager()
     */
    public function translate($message, $textDomain = 'default', $locale = null)
    {
-       $locale      = $locale ?: $this->getLocale();
+       $locale      = $locale ?? $this->getLocale();
        $translation = $this->getTranslatedMessage($message, $locale, $textDomain);
``` 
https://github.com/laminas/laminas-i18n/pull/117/files#diff-617dd26fad700da3ee06fad2afafa9652a82a157416b914efbada1998c38d478L348

#### Previous behavior

`\Laminas\I18n\Translator\Translator::translate()` returns translated message using default locale when empty string is passed as locale 

#### Current behavior

`\Laminas\I18n\Translator\Translator::translate()` returns untranslated message when empty string is passed as locale.

#### How to reproduce

pass empty string as locale to `\Laminas\I18n\Translator\Translator::translate()`

### PR DESCRIPTION
**Pull request by kersysgediminas**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Submitting failing unit test for issue https://github.com/laminas/laminas-i18n/issues/135

### CODE REVIEW

**Review by gsteel — APPROVED** [review]
Thanks for the failing test @kersysgediminas and the detailed issue report.
